src/components/CreateCourse/CreateCourse.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { useState } from 'react';
+import CreateTitle from './components/CreateTitle/CreateTitle';
+import CreateCourseBody from './components/CreateCourseBody/CreateCourseBody';
+import CreateAuthor from './components/CreateAuthor/CreateAuthor';
+import AddDurationNewCourse from './components/AddDurationNewCourse/AddDurationNewCourse';
+import AddedAuthorsToCourse from './components/AddedAuthorsToCourse/AddedAuthorsToCourse';
+import AutorsList from './components/AutorsList/AutorsList';
+
+import { generateUUID } from '../../helpers/generateUUID';
+import { isAllFieldesFilled } from '../../helpers/isAllFieldsFilled';
+import { Course, Author } from '../../interfaces/index';
+
+import './createCourse.css';
+
+type Props = {
+	courses: Course[];
+	setCourses: (courses: Course[]) => void;
+	authorsList: Author[];
+	setAuthorsList: (authors: Author[]) => void;
+	setRenderedComponent: (component: string) => void;
+	renderedComponent: string;
+};
+
+function CreateCourse(props: Props) {
+	const [authors, setAuthors] = useState<string[]>([]);
+
+	function submitHandler(e: React.SyntheticEvent) {
+		e.preventDefault();
+
+		const target = e.target as typeof e.target & {
+			courseTitle: { value: string };
+			courseDescription: { value: string };
+			duration: { value: string };
+		};
+
+		const newCourse: Course = {
+			id: generateUUID(),
+			title: target.courseTitle.value,
+			description: target.courseDescription.value,
+			creationDate: new Date().toLocaleDateString('en-GB'),
+			duration: Number(target.duration.value),
+			authors: authors,
+		};
+
+		isAllFieldesFilled(newCourse);
```

Comment:
You are throwing an error here and not handling it in any way. 
If you want to use error here then at least wrap this portion in `try catch`.

The better way will be to return false from `isAllFieldesFilled` if not all fields are filled and stop executing the handler by using `return`

```
if (!isAllFieldesFilled(newCourse)) {
  return
} else {
  <do some stuff>
}
```

Reply:
done